processor/elasticinframetricsprocessor/processor.go

```diff
@@ -54,63 +61,52 @@ func newProcessor(set processor.Settings, cfg *Config) *ElasticinframetricsProce
 			scopeMetric := resourceMetric.ScopeMetrics().At(scopeIndex)
+			remappedMetricsSlice := pmetric.NewMetricSlice()
 			for _, r := range p.remappers {
-				r.Remap(scopeMetric, scopeMetric.Metrics(), rm)
+				r.Remap(scopeMetric, remappedMetricsSlice, rm)
+			}
+			if remappedMetricsSlice.Len() > 0 {
+				remappedScopeMetric := remappedResourceMetrics.ScopeMetrics().AppendEmpty()
+				remappedMetricsSlice.MoveAndAppendTo(remappedScopeMetric.Metrics())
+				hasRemappedMetrics = true
 			}
 		}
 	}
-	// drop_original=true will keep only the metrics that have been remapped based on the presense of OTelRemappedLabel label.
-	// See  https://github.com/elastic/opentelemetry-lib/blob/6d89cbad4221429570107eb4a4968cf8a2ff919f/remappers/common/const.go#L31
-	if p.cfg.DropOriginal {
-		newMetic := pmetric.NewMetrics()
-		for resIndex := range md.ResourceMetrics().Len() {
-			resourceMetric := md.ResourceMetrics().At(resIndex)
-			rmNew := newMetic.ResourceMetrics().AppendEmpty()
-
-			// We need to copy Resource().Attributes() because those inlcude additional attributes of the metrics
-			resourceMetric.Resource().Attributes().CopyTo(rmNew.Resource().Attributes())
-			for scopeIndex := range resourceMetric.ScopeMetrics().Len() {
-				scopeMetric := resourceMetric.ScopeMetrics().At(scopeIndex)
-				rmScope := rmNew.ScopeMetrics().AppendEmpty()
-
-				// Iterate over the metrics
-				for metricIndex := range scopeMetric.Metrics().Len() {
-					metric := scopeMetric.Metrics().At(metricIndex)
+	var errs []error
+	if !p.cfg.DropOriginal {
+		errs = append(errs, p.next.ConsumeMetrics(ctx, md))
+	}
+	if hasRemappedMetrics {
+		ecsCtx := client.NewContext(ctx, withMappingMode(client.FromContext(ctx), "ecs"))
+		errs = append(errs, p.next.ConsumeMetrics(ecsCtx, remappedMetrics))
+	}
 
-					if metric.Type() == pmetric.MetricTypeGauge {
-						for dataPointIndex := range metric.Gauge().DataPoints().Len() {
-							if oTelRemappedLabel, ok := metric.Gauge().DataPoints().At(dataPointIndex).Attributes().Get(OTelRemappedLabel); ok {
-								if oTelRemappedLabel.Bool() {
-									metric.CopyTo(rmScope.Metrics().AppendEmpty())
-								}
-							}
-						}
-					} else if metric.Type() == pmetric.MetricTypeSum {
-						for dataPointIndex := range metric.Sum().DataPoints().Len() {
-							if oTelRemappedLabel, ok := metric.Sum().DataPoints().At(dataPointIndex).Attributes().Get(OTelRemappedLabel); ok {
-								if oTelRemappedLabel.Bool() {
-									resourceMetric.Resource().Attributes().CopyTo(rmNew.Resource().Attributes())
-									metric.CopyTo(rmScope.Metrics().AppendEmpty())
-								}
-							}
-						}
-					}
+	return errors.Join(errs...)
+}
 
-				}
-			}
-		}
-		return newMetic, nil
+func withMappingMode(info client.Info, mode string) client.Info {
+	return client.Info{
+		Addr:     info.Addr,
+		Auth:     info.Auth,
+		Metadata: client.NewMetadata(map[string][]string{"x-elastic-mapping-mode": {mode}}),
```

Comment:
It looks like there's no way to retain the existing metadata i.e. we're overriding it here. Could this be an issue? @axw are we planning to add something like the project id to the metadata that would get removed by setting the mapping mode?

Reply:
Urgh :) Yeah we will be setting that in client metadata, and yeah that is an issue indeed. I believe we would need to add a method to client.Metadata to get its keys.

Reply:
Or a method to create a new metadata by taking in old metadata and a map - it might be more efficient and play well with the immutability of the metadata.

Reply:
As discussed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38114, we should change this to use a scope attribute rather than client metadata.